x/programs/examples/util.go

```diff
@@ -5,29 +5,62 @@ package examples
 
 import (
 	"context"
+	"encoding/binary"
 	"os"
 
 	"github.com/ava-labs/avalanchego/database/memdb"
+	"github.com/ava-labs/hypersdk/consts"
 	"github.com/ava-labs/hypersdk/crypto/ed25519"
 	"github.com/ava-labs/hypersdk/state"
 	"github.com/ava-labs/hypersdk/x/programs/runtime"
+	"github.com/near/borsh-go"
 )
 
-func newKeyPtr(ctx context.Context, key ed25519.PublicKey, rt runtime.Runtime) (int64, error) {
-	ptr, err := rt.Memory().Alloc(ed25519.PublicKeyLen)
+// newPtr allocates memory and writes [bytes] to it.
+// If [prependLength] is true, it prepends the length of [bytes] as a uint32 to [bytes].
+// It returns the pointer to the allocated memory.
+func newPtr(ctx context.Context, item interface{}, rt runtime.Runtime, prependLength bool) (int64, error) {
+	bytes, err := serializeToBytes(item)
+	if err != nil {
+		return 0, err
+	}
+
+	amountToAllocate := uint64(len(bytes))
+	writeBytes := bytes
+
+	if prependLength {
+		amountToAllocate += consts.Uint32Len
+		writeBytes = marshalArg(bytes)
+	}
+
+	ptr, err := rt.Memory().Alloc(amountToAllocate)
```

Comment:
I feel like these should be separate steps see below.

https://github.com/ava-labs/hypersdk/pull/631/files#diff-46bbbce39f0b86c15aea460cf21fa799bd6fabad826e7ea351c7577e13997ac9R206-R228